### DISCUSSION TITLE
Proposed new features for the "Chapter" feature in Chessforge

### DISCUSSION THREAD

**Discussion by akaKarthick** [czbar/ChessForge#730] [General]
My first suggestion is to implement keyboard **shortcuts** for selecting games/exercises, selecting all games/exercises, and deleting games/exercises. The following keyboard shortcuts are common and would be easy for users to remember:

- Select games/exercises: `Ctrl+Right click`

- Select all games/exercises: `Ctrl+A`

- Delete games/exercises: `Delete`

My second suggestion is to add a **drag-and-drop** option for moving games/exercises up, down, and deleting them. This would be a more intuitive and user-friendly way to perform these operations than the current method of using the context menu.

For example, to move a game to a different location in the chapter, the user could simply click and drag the game name to the desired location. To delete a game, the user could simply drag and drop the game name into a "Delete" folder.

In addition to the above suggestions, I would also like to suggest the implementation of a **"Recycle Bin"** feature (Chapter or icon) in Chessforge. This would allow users to recover deleted games/exercises if they wish.

Deleted games/exercises would be stored in the Recycle Bin for a certain period of time, such as 30 days. After this period of time, the deleted games/exercises would be permanently deleted. ( keep #657 this in mind )

I know that a Recycle Bin feature is crazy, but it would prevent users from accidentally deleting games/exercises that they do not want to delete.

I just sharing my thoughts with you. You pick and implement what you want.

**Comment by czbar** [comment]
As for the **drag-and-drop**, nearly a year ago, I started writing a technical spec for it https://github.com/czbar/ChessForge/wiki/Chapters-View-mouse-events but never got around to actually doing it. It's quite a bit of work and there was always something more urgent to do. So, yes, it is pending...

As for the **keyboard shortcuts**, `Delete` to delete a single game is easy, of course, but selecting multiple entities in the Chapters view may possibly be confusing from the GUI's point of view. So I am not too sure at this point... Note that I have re-worked the multiple selection dialog (in the next version 1.3.3) that will be used, in particular, when deleting multiple games.

![image](https://github.com/czbar/ChessForge/assets/7270924/cc73e86d-346e-4162-b661-a7260378b1e3)

It allows you to expand/collapse chapters (by double-clicking) and to select what you need.

As for the **Recycle Bin** ... I quite like the idea!  It is relatively easy to implement.  (I have implemented Undo for many operations and that is much harder that something like the Recycle Bin.)
